edge-modules/Relayer/Settings.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Relayer
+{
+    using System;
+    using System.IO;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using Newtonsoft.Json.Serialization;
+
+    [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+    public class Settings
+    {
+        static readonly Lazy<Settings> DefaultSettings = new Lazy<Settings>(
+            () =>
+            {
+                IConfiguration configuration = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("config/settings.json", optional: true)
+                    .AddEnvironmentVariables()
+                    .Build();
+
+                return new Settings(
+                    configuration.GetValue("transportType", TransportType.Amqp_Tcp_Only),
+                    configuration.GetValue("outputName", "output1"));
+            });
+
+        Settings(
+            TransportType transportType,
+            string outputName)
+        {
+            this.TransportType = transportType;
+            this.OutputName = outputName;
```

Comment:
Recommend using Preconditions.Check statements to validate arguments not null. For an example, look at ModuleRestarter.